src/inmanta/parser/plyInmantaParser.py

```diff
@@ -871,8 +884,14 @@ def p_index_lookup(p: YaccProduction) -> None:
 
 
 def p_short_index_lookup(p: YaccProduction) -> None:
-    "index_lookup : attr_ref '[' param_list ']'"
+    "index_lookup : expression '[' param_list ']'"
     attref = p[1]
+    if not isinstance(attref, AttributeReference):
+        raise ParserException(
+            attref.location,
+            attref,
+            f"Short index lookup must start with an attribute reference (`<expression>.<id>`), but got `{repr(attref)}`",
```

Comment:
I think you would get this when you try to do a list index operation? Perhaps this error message would just be more confusing to a user that thinks we support list indexing?

Reply:
Ah no, `param_list` means kwargs...

Reply:
True, what would you say instead?

Reply:
I think my second comment invalidates the first, do you agree?